shell/platform/android/platform_view_android_jni_impl.h

```diff
@@ -0,0 +1,61 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_ANDROID_PLATFORM_VIEW_ANDROID_JNI_IMPL_H_
+#define FLUTTER_SHELL_PLATFORM_ANDROID_PLATFORM_VIEW_ANDROID_JNI_IMPL_H_
+
+#include "flutter/fml/platform/android/jni_weak_ref.h"
+#include "flutter/shell/platform/android/platform_view_android_jni.h"
+
+namespace flutter {
+
+class PlatformViewAndroidJniImpl final : public PlatformViewAndroidJni {
+ public:
+  PlatformViewAndroidJniImpl(fml::jni::JavaObjectWeakGlobalRef java_object);
+
+  ~PlatformViewAndroidJniImpl() override;
+
+  void FlutterViewHandlePlatformMessage(
+      fml::RefPtr<flutter::PlatformMessage> message,
+      int responseId) override;
+
+  void FlutterViewHandlePlatformMessageResponse(
+      int responseId,
+      std::unique_ptr<fml::Mapping> data) override;
+
+  void FlutterViewUpdateSemantics(std::vector<uint8_t> buffer,
+                                  std::vector<std::string> strings) override;
+
+  void FlutterViewUpdateCustomAccessibilityActions(
+      std::vector<uint8_t> actions_buffer,
+      std::vector<std::string> strings) override;
+
+  void FlutterViewOnFirstFrame() override;
+
+  void FlutterViewOnPreEngineRestart() override;
+
+  void SetCurrentSurfaceTexture(
+      fml::jni::JavaObjectWeakGlobalRef& surface_texture);
+
+  void SurfaceTextureAttachToGLContext(int textureId) override;
+
+  void SurfaceTextureUpdateTexImage() override;
+
+  void SurfaceTextureGetTransformMatrix(SkMatrix& transform) override;
+
+  void SurfaceTextureDetachFromGLContext() override;
+
+ private:
+  // Reference to FlutterJNI.
+  const fml::jni::JavaObjectWeakGlobalRef java_object_;
+
+  // Reference to the current Surface texture object.
+  fml::jni::JavaObjectWeakGlobalRef surface_texture_;
```

Comment:
Why does this object hold another reference to the same `SurfaceTexture` held by an `AndroidExternalTextureGL` instance?

If the goal is to centralize all native->Java calls into one interface, then can the `SurfaceTexture` reference be passed as an argument instead of duplicating state here?

Or would it make sense to split this into separate JNI wrapper interfaces for `FlutterJNI` and `SurfaceTexture`?

Reply:
The issue is that `SurfaceTexture` requires the Android toolchain. The idea is that the new interface doesn't require the Android toolchain.  I had to downcast  `jni_facade_` in the constructor, so I can access the method.  

Splitting into a separate wrapper is possible if this workaround doesn't look good.

Reply:
I would prefer moving the `SurfaceTexture` methods into a separate wrapper

Reply:
What about we use the texture id (which is `int64_t`) to lookup the `SurfaceTexture` in Java?  This way, we can keep everything under the same interface.  I'm afraid that if we create a separate wrapper, then it may diminish the value of having a centralized interface.

Reply:
I'm concerned that doing the lookup would add unnecessary overhead to frequently called APIs like `UpdateTexImage`.

Can the `SurfaceTexture` JNI invocations and method ID lookups be moved into `android_external_texture_gl.cc`?  That file is already dependent on JNI and OpenGL ES APIs.

Reply:
What does https://github.com/flutter/engine/pull/18903/commits/c72c3eb4b1449a2ed9f1a24fdc24c1486156583c look like?

Reply:
What does 19aa4e7 look like?

Reply:
SGTM - hiding the JNI object reference behind a `std::any` seems like a reasonable way to go if there is a requirement to make all JNI calls through one interface but that interface can not use JNI types.